framework/model/source/k8s/metric_source.go

```diff
@@ -0,0 +1,306 @@
+package k8s
+
+import (
+	"context"
+	cmap "github.com/orcaman/concurrent-map"
+	prometheusApi "github.com/prometheus/client_golang/api"
+	prometheus "github.com/prometheus/client_golang/api/prometheus/v1"
+	prometheusModel "github.com/prometheus/common/model"
+	log "github.com/sirupsen/logrus"
+	"istio.io/api/networking/v1alpha3"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/rest"
+	"reflect"
+	bootconfig "slime.io/slime/framework/apis/config/v1alpha1"
+	"slime.io/slime/framework/bootstrap"
+	"slime.io/slime/framework/controllers"
+	"slime.io/slime/framework/model"
+	"slime.io/slime/framework/model/module"
+	"slime.io/slime/framework/util"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+)
+
+type MetricSource struct {
+	name                     string
+	gvks                     []schema.GroupVersionKind
+	metricUpdateCheckHandler func(event model.WatcherEvent) map[string]*bootconfig.Prometheus_Source_Handler
```

Comment:
metricUpdateCheckHandler check handler?
should be better for `eventHandlers` or `watcherEventHandlers`